### PR TITLE
Add ses limit check job

### PR DESCRIPTION
**Pull request by jdmulloy**

### CODE REVIEW

File: devops/jobs/CheckSesLimits.groovy [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                            command(
+                                dslFactory.readFileFromWorkspace("devops/resources/check-ses-limits.sh")
+                            )
+
+                        }
+
+                    }
+
+                    publishers {
+                        mailer(extraVars.get('NOTIFY_ON_FAILURE','devops+non-critical@edx.org'), false, false)
```
Comment (feanil) [inline devops/jobs/CheckSesLimits.groovy]: You have a conditional above that alwas sets this key in the extraVars dict, so I don't think you need a default here.  You should just assert that the NOTIFY_ON_FAILURE key is in the dict.

File: devops/jobs/CheckSesLimits.groovy [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                }
+
+                if (!configuration.containsKey('CONFIGURATION_SECURE_REPO')) {
+                        extraVars['CONFIGURATION_SECURE_REPO'] = "git@github.com:edx/${deployment}-secure.git"
+                }
+                else {
+                        extraVars['CONFIGURATION_SECURE_REPO'] = configuration['CONFIGURATION_SECURE_REPO']
+                }
+
+                properties {
```
Comment (feanil) [inline devops/jobs/CheckSesLimits.groovy]: Remove this.  Not sure why we need this.

File: devops/jobs/CheckSesLimits.groovy [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+                wrappers {
+                    credentialsBinding {
+                        file('AWS_CONFIG_FILE','tools-edx-jenkins-aws-credentials')
+                        def variable = "ses-role-${deployment}-arn"
+                        string('ROLE_ARN', variable)
+                    }
+                  }
+
+                def crit_threshold = extraVars.get('CRIT_THRESHOLD','')
```
Comment (feanil) [inline devops/jobs/CheckSesLimits.groovy]: Don't set default values, assert that value is a number at the top.

File: devops/jobs/CheckSesLimits.groovy [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+                def warn_threshold = extraVars.get('WARN_THRESHOLD','')
```
Comment (feanil) [inline devops/jobs/CheckSesLimits.groovy]: Don't set default values, assert that value is a number at the top.

File: devops/jobs/CheckSesLimits.groovy [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+import static org.edx.jenkins.dsl.DevopsTasks.common_multiscm
+
+class CheckSesLimits {
+    public static def job = { dslFactory, extraVars ->
+        assert extraVars.containsKey('DEPLOYMENTS') : "Please define DEPLOYMENTS. It should be a list of strings."
+        assert extraVars.containsKey('CRIT_THRESHOLD') : "Please define CRIT_THRESHOLD. It should be a percentage expressed as a float between 0 to 100."
+        assert extraVars.containsKey('WARN_THRESHOLD') : "Please define WARN_THRESHOLD. It should be a percentage expressed as a float between 0 to 100."
+        assert extraVars.containsKey('NOTIFY_ON_FAILURE') : "Please define NOTIFY_ON_FAILURE."
+        assert !(extraVars.get('DEPLOYMENTS') instanceof String) : "Make sure DEPLOYMENTS is a list and not a string"
+        assert (extraVars.get('CRIT_THRESHOLD') instanceof Integer || extraVars.get('CRIT_THRESHOLD') instanceof Double) : "Make sure CRIT_THRESHOLD is a number between 0 and 100"
```
Comment (feanil) [inline devops/jobs/CheckSesLimits.groovy]: Check to see if it's an instance of `Number` instead.

https://docs.oracle.com/javase/7/docs/api/java/lang/Number.html

Comment (jdmulloy) [inline devops/jobs/CheckSesLimits.groovy]: Ok. I've changed it. When I typed Number into vim it didn't change color like the Integer and Double types did.

File: devops/jobs/CheckSesLimits.groovy [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    Variables without defaults are marked (required)
+
+    Variables consumed for this job:
+        * NOTIFY_ON_FAILURE: (Required - email address)
+        * SECURE_GIT_CREDENTIALS: (Required - jenkins name of git credentials)
+        * CRIT_THRESHOLD: (Required - int or float between 0 and 100)
+        * WARN_THRESHOLD: (Required - int or float between 0 and 100)
+        * DEPLOYMENTS: (Required)
+          deployment_name: (Required)
+            regions: (Required - space seperated list of aws regions)
```
Comment (hchen15) [inline devops/jobs/CheckSesLimits.groovy]: small typo for separated but otherwise looks good

Comment (jdmulloy) [inline devops/jobs/CheckSesLimits.groovy]: Fixed

**Pull request merged**